### PR TITLE
Lesson04 - mailroom2.py

### PR DESCRIPTION
**Pull request by wpucilow**

### CODE REVIEW

File: students/Wieslaw_Pucilowski/lesson04/mailroom2.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        print('{:<30}{}{:>18.2f}{:>11}{}{:>17.2f}'.format(k[0]+' '+k[1], ' $', sum(v), len(v), ' $', sum(v)/len(v)))
+
+def menu_selection(prompt, dispatch_dict):
+    while True:
+        response = input(prompt)
+        if dispatch_dict.get(response, "default") == "default":
+            print(response,wrong_option)
+            continue
+        elif dispatch_dict[response]() == "exit menu": # mind dispatch_dict[response] return function and following () call it
+            break
```
[the remark this answers is not in the document]
Comment (wpucilow) [reply students/Wieslaw_Pucilowski/lesson04/mailroom2.py]: I can't see how can I break menu while loop by the dispatch function in this configuration.

[the remark this answers is not in the document]
Comment (robalford) [reply students/Wieslaw_Pucilowski/lesson04/mailroom2.py]: If you place the call to the dispatch function inside the while loop, then your program will, at least temporarily, exit the while loop to the body of the function called when it is encountered. So you could place a call to `sys.exit(0)` in the `quit` function and the program will terminate at that point. Then you wouldn't need the conditional check for the `quit` function at all

**Comment by robalford** [comment]
@wpucilow don't forget to submit these assignments on the course website as well so we can grade. Thanks!

File: students/Wieslaw_Pucilowski/lesson04/mailroom2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wpucilow) [reply students/Wieslaw_Pucilowski/lesson04/mailroom2.py]: I have just checked and if I do sys.exit() inside quit() function, then after choosing 'q' option my script terminates wherever I am in Main menu or Sub menu. My goal was when choosing 'q' to terminate the script if I'm in Main menu and to go to Main menu if I'm in Sub menu.

**Comment by wpucilow** [comment]
I have uploaded updated scripts to github

**Pull request merged**